spot_driver/test/src/conversions/test_robot_state.cpp

```diff
@@ -353,14 +353,119 @@ TEST(RobotStateConversions, TestGetOdomTwist) {
   const auto is_using_vision = false;
   const auto out = getOdomTwist(robot_state, clock_skew, is_using_vision);
 
+  // THEN this fails as there is no TransformsSnapshot
+  ASSERT_THAT(out.has_value(), IsFalse());
+}
+
+TEST(RobotStateConversions, TestGetOdomTwistBodyAtOdom) {
+  // GIVEN a RobotState that contains info about the velocity of the body in the odom frame
+  ::bosdyn::api::RobotState robot_state;
+  google::protobuf::Timestamp timestamp;
+  timestamp.set_seconds(99);
+  timestamp.set_nanos(0);
+  addAcquisitionTimestamp(robot_state.mutable_kinematic_state(), timestamp);
+
+  // GIVEN the odom frame is the root of the frame tree
+  addRootFrame(robot_state.mutable_kinematic_state()->mutable_transforms_snapshot(), "odom");
+  // GIVEN the body frame is at the odom frame
+  addTransform(robot_state.mutable_kinematic_state()->mutable_transforms_snapshot(), "body", "odom", 0.0, 0.0, 0.0, 1.0,
+               0.0, 0.0, 0.0);
+
+  auto* velocity_linear = robot_state.mutable_kinematic_state()->mutable_velocity_of_body_in_odom()->mutable_linear();
+  velocity_linear->set_x(1.0);
+  velocity_linear->set_y(2.0);
+  velocity_linear->set_z(3.0);
+  auto* velocity_angular = robot_state.mutable_kinematic_state()->mutable_velocity_of_body_in_odom()->mutable_angular();
+  velocity_angular->set_x(1.0);
+  velocity_angular->set_y(2.0);
+  velocity_angular->set_z(3.0);
+
+  // GIVEN some nominal clock skew
+  google::protobuf::Duration clock_skew;
+  clock_skew.set_seconds(1);
+
+  // WHEN we create a TwistWithCovarianceStamped ROS message
+  const bool using_vision = false;
+  const auto out = getOdomTwist(robot_state, clock_skew, using_vision);
+
   // THEN this succeeds
   ASSERT_THAT(out.has_value(), IsTrue());
 
-  // THEN the output twist matches the velocity in the robot state
+  // THEN the output twist matches the velocity in the robot state.
   EXPECT_THAT(out->twist.twist, GeometryMsgsTwistEq(1.0, 2.0, 3.0, 1.0, 2.0, 3.0));
   EXPECT_THAT(out->header, ClockSkewIsAppliedToHeader(timestamp, clock_skew));
 }
 
+TEST(RobotStateConversions, TestGetOdomTwistLinearTransformationLinearVelocity) {
+  // GIVEN a RobotState that contains info about the velocity of the body in the odom frame
+  ::bosdyn::api::RobotState robot_state;
+  google::protobuf::Timestamp timestamp;
+  timestamp.set_seconds(99);
+  timestamp.set_nanos(0);
+  addAcquisitionTimestamp(robot_state.mutable_kinematic_state(), timestamp);
+
+  // GIVEN the odom frame is the root of the frame tree
+  addRootFrame(robot_state.mutable_kinematic_state()->mutable_transforms_snapshot(), "odom");
```

Comment:
@khughes-bdai nit^N!: having a local alias may improve readability a bit:
```suggestion
  auto *transforms_snapshot = robot_state.mutable_kinematic_state()->mutable_transforms_snapshot();
  addRootFrame(transforms_snapshot, "odom");
```
Same everywhere else.